### PR TITLE
objectstore: simplify/consolidate collection_list()

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by athanatos** [comment]
lgtm!

File: src/os/GenericObjectMap.cc [hunk 1]
```diff
@@ -1099,7 +1099,14 @@ int GenericObjectMap::list_objects(const coll_t &cid, ghobject_t start, int max,
       break;
     }
 
+    if (header.oid >= end) {
+      if (next)
+	*next = ghobject_t::get_max();
```
Comment (yuyuyu101) [inline src/os/GenericObjectMap.cc]: It looks a little confuse to me, is "end" must be "ghobject_t::get_max()"? 
It should be " *next = header.oid;" more reasonable?

Comment (liewegas) [inline src/os/GenericObjectMap.cc]: The semantic is that once you hit end next == max (and you know you're done)

Comment (yuyuyu101) [inline src/os/GenericObjectMap.cc]: got!

**Comment by liewegas** [comment]
will rebase and reopen a version for qa

**Pull request closed without merging**